src/hooks/useHandleTransfer.tsx

```diff
@@ -794,8 +752,9 @@ async function injective(
       asset,
       transferAmountParsed.toString(),
       targetChain,
-      targetAddress,
-      feeParsed.toString()
+      additionalPayload?.receivingContract || targetAddress,
+      feeParsed.toString(),
+      additionalPayload?.payload
```

Comment:
Question:
Why do we pass an additional `additionalPayload?.payload` param here, and in others `transferFromX` methods?

Reply:
So if later we need to send an additional payload (or send a tx to a different contract) on any chain we'll just need to add the condition into the maybeAdditionalPayload function 👌🏼